src/test/java/com/endava/internship/collections/StudentListTest.java

```diff
@@ -0,0 +1,551 @@
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class StudentListTest {
+    List<Student> studentList;
+
+    @BeforeEach
+    void setUp() {
+        studentList = new StudentList<>();
+        Student st1 = new Student("Unu", LocalDate.of(2001, 1, 1), "Student ONE");
+        Student st2 = new Student("Doi", LocalDate.of(2004, 1, 1), "Student TWO");
+        Student st3 = new Student("Trei", LocalDate.of(2006, 1, 1), "Student TREE");
+        studentList.add(st1);
+        studentList.add(st2);
+        studentList.add(st3);
+    }
+
+    @Test
+    void shouldThrowIllegalArgumentExceptionWithInvalidParameter() {
+        Throwable exceptionThatWasThrown = assertThrows(IllegalArgumentException.class, () -> new StudentList<Student>(-1));
+        assertEquals(exceptionThatWasThrown.getMessage(), "Illegal Capacity: " + "-1");
+    }
+
+    @Test
+    void shouldCreateAListWithDefaultCapacity() {
+        List<Student> studentList = new StudentList<>(8);
+
+        assertNotNull(studentList);
+    }
+
+    @Test
+    void shouldCreatListWithLengthFifteen() {
+        List<Student> studentList = new StudentList<>(15);
+
+        assertNotNull(studentList);
+    }
+
+    @Test
+    void shouldReturnTheSizeEqualsToTree() {
+        int listSize = studentList.size();
+
+        assertEquals(listSize, 3);
+    }
+
+    @Test
+    void shouldReturnTrueWhenTheListHasNoElements() {
+        List<Student> studentList = new StudentList<>();
+        assertTrue(studentList.isEmpty());
+    }
+
+    @Test
+    void shouldReturnFalseWhenTheListHasElements() {
+        assertFalse(studentList.isEmpty());
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideStudentForContains")
+    void contains_ShouldReturnExpectedBooleans(Student student, boolean expected) {
+        assertEquals(expected, studentList.contains(student));
+    }
+
+    private static Stream<Arguments> provideStudentForContains() {
+        Student st1 = new Student("Unu", LocalDate.of(2001, 1, 1), "Student ONE");
+        Student st4 = new Student("Patru", LocalDate.of(2006, 1, 1), "Student PATRU");
+        return Stream.of(
+                Arguments.of(st1, true),
+                Arguments.of(st4, false),
+                Arguments.of(null, false));
+    }
+
+    @Nested
+    class testAllMethodsOfTheInnerClassIterator {
```

Comment:
class name should start with capital letter

Reply:
Done. Thank you.